React/Modules/RCTUIManager.m

```diff
@@ -723,6 +727,20 @@ - (void)_amendPendingUIBlocksWithStylePropagationUpdateForShadowView:(RCTShadowV
   }
 }
 
+- (void)addUIManagerObserver:(id<RCTUIManagerObserver>)observer
+{
+  [_uiManagerObserversLock lock];
```

Comment:
The case Im worried about is:
UIManager batchDidComplete -> dispatch async the ui operations block
NativeAnimatedModule batchDidComplete -> dispatch async addObserver.

In this case the observer would get added after operations are executed. It's preferable not to assume the order in which the batchDidComplete methods get executed in.